### PR TITLE
#1739 Improve Copy Legend Functionality 

### PR DESCRIPTION
**Pull request by MohamedAsli**
#1739 
This PR enhances the functionality for copying legend views between documents in Revit, and handles some corner cases with catching exception

#### Key Improvements:
- **Progress Bar** 
- **Error Handling:** Improved handling and detailed logging of errors and skipped elements
- **Transaction Group** 
- **User Alerts:** Added clear success/failure messages and alerts for critical issues
- **Logging:** Displays logs after the script finishes

This is my first contribution, so feel free to point out any areas for improvement

### CODE REVIEW

**Comment by MohamedAsli** [comment]
tested on revit 20 and 24

File: extensions/pyRevitTools.extension/pyRevit.tab/Drawing Set.panel/views.stack/Legends.pulldown/Copy Legends to Other Documents.pushbutton/script.py [hunk 1]
```diff
@@ -20,79 +19,112 @@ def OnDuplicateTypeNamesFound(self, args):
...
-                if isinstance(el, DB.Element) and el.Category:
-                    elements_to_copy.append(el.Id)
-                else:
-                    logger.debug('Skipping element: %s', el.Id)
-            if not elements_to_copy:
-                logger.debug('Skipping empty view: %s',
-                             revit.query.get_name(src_legend))
+legends = forms.select_views(title='Select Legend Views',
+                             filterfunc=lambda x: x.ViewType == DB.ViewType.Legend,
+                             use_selection=True)
```
Comment (jmcouffin) [inline extensions/pyRevitTools.extension/pyRevit.tab/Drawing Set.panel/views.stack/Legends.pulldown/Copy Legends to Other Documents.pushbutton/script.py:24]: I would go like this for the progress bar
```python
legends_nb = len(legends) * len(open_docs) # total number of legends to copy
count = 0

skipped_docs = []
if open_docs and legends:
    with forms.ProgressBar(title='Processing',
                           cancellable=True) as pb:
        pb.update_progress(0, legends_nb)

'''
processing the legends
'''
                    count += 1
                    pb.update_progress(count, legends_nb)
```

File: extensions/pyRevitTools.extension/pyRevit.tab/Drawing Set.panel/views.stack/Legends.pulldown/Copy Legends to Other Documents.pushbutton/script.py [hunk 2]
```diff
@@ -20,79 +19,112 @@ def OnDuplicateTypeNamesFound(self, args):
...
+                        new_name = src_name
+                        while new_name in all_legend_names:
+                            count += 1
+                            new_name = src_name + ' (Duplicate %s)' % count
+                            logger_messages.append(
+                                'Legend name already exists. Renaming to: {}'.format(new_name))
+                        revit.update.set_name(dest_view, new_name)
+                        dest_view.Scale = src_legend.Scale
+
+                    if copy_transaction.status == DB.TransactionStatus.Committed:
```
Comment (jmcouffin) [inline extensions/pyRevitTools.extension/pyRevit.tab/Drawing Set.panel/views.stack/Legends.pulldown/Copy Legends to Other Documents.pushbutton/script.py:108]: it did not work in my tests

> The managed object is not valid

File: extensions/pyRevitTools.extension/pyRevit.tab/Drawing Set.panel/views.stack/Legends.pulldown/Copy Legends to Other Documents.pushbutton/script.py (hunk 1 above)
Comment (jmcouffin) [reply extensions/pyRevitTools.extension/pyRevit.tab/Drawing Set.panel/views.stack/Legends.pulldown/Copy Legends to Other Documents.pushbutton/script.py:24]: you know the number of legends you want to copy and in how many document.
therefore, using indeterminate state for the progress bar does not make sense

File: extensions/pyRevitTools.extension/pyRevit.tab/Drawing Set.panel/views.stack/Legends.pulldown/Copy Legends to Other Documents.pushbutton/script.py [hunk 3]
```diff
@@ -20,79 +19,112 @@ def OnDuplicateTypeNamesFound(self, args):
...
+                        doc_processed = True
+
+    if doc_processed:
+        if skipped_docs:
```
Comment (jmcouffin) [inline extensions/pyRevitTools.extension/pyRevit.tab/Drawing Set.panel/views.stack/Legends.pulldown/Copy Legends to Other Documents.pushbutton/script.py:112]: this looks redundant, 
first list you processed the title
second you don't 
third you process the titles @MohamedAsli

**Comment by MohamedAsli** [comment]
@jmcouffin any feedback ?

File: extensions/pyRevitTools.extension/pyRevit.tab/Drawing Set.panel/views.stack/Legends.pulldown/Copy Legends to Other Documents.pushbutton/script.py (hunk 3 above)
Comment (jmcouffin) [reply extensions/pyRevitTools.extension/pyRevit.tab/Drawing Set.panel/views.stack/Legends.pulldown/Copy Legends to Other Documents.pushbutton/script.py:112]: @MohamedAsli 🫴

File: extensions/pyRevitTools.extension/pyRevit.tab/Drawing Set.panel/views.stack/Legends.pulldown/Copy Legends to Other Documents.pushbutton/script.py (hunk 2 above)
Comment (jmcouffin) [reply extensions/pyRevitTools.extension/pyRevit.tab/Drawing Set.panel/views.stack/Legends.pulldown/Copy Legends to Other Documents.pushbutton/script.py:108]: @MohamedAsli 🫴

**Comment by jmcouffin** [comment]
> @jmcouffin any feedback ?
@MohamedAsli 

yes, up there ☝️ 
but I made a mistake as I deleted the develop-4 branch, I forgot to rebase your PR from develop branch.
Would you mind making a new PR from the develop branch and review my comments ☝️

**Comment by jmcouffin** [comment]
@MohamedAsli forgot to press 'Submit The Review' 😄

**Comment by jmcouffin** [comment]
And I did review back in june 😨

**Comment by GeorgePottsSDS** [comment]
Is this sorted now? I raised the inital query :D Still waiting for the update haha

**Pull request closed without merging**